### PR TITLE
serde config

### PR DESCRIPTION
**Pull request by drunkirishcoder**
`config rs` doesn't add much value. that dependency is removed. instead use `serde` and `toml` crates directly.

### CODE REVIEW

File: core/src/config.rs [hunk 1]
```diff
@@ -28,82 +27,120 @@ impl<'de> Deserialize<'de> for MacAddr {
...
 
-/// Runtime settings.
+/// Deserializes a duration from seconds expressed as `u64`.
+pub fn duration_from_secs<'de, D>(deserializer: D) -> std::result::Result<Duration, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let secs = u64::deserialize(deserializer)?;
+    Ok(Duration::from_secs(secs))
+}
```
Comment (clinedome) [inline core/src/config.rs]: there's no magic that causes this to be called for any default deserialize of `Duration`, right? I see that we're explicitly specifying it down below, which is good.

Comment (drunkirishcoder) [reply core/src/config.rs]: nope. no magic unfortunately. has to be explicitly referenced by the `serde` attribute.

File: core/src/config.rs [hunk 2]
```diff
@@ -28,82 +27,120 @@ impl<'de> Deserialize<'de> for MacAddr {
...
     /// processes together in a multi-process setup; and allow them to share
     /// the same memory regions. The default value is the `app_name`. Each
     /// process works independently.
+    #[serde(default)]
     pub app_group: Option<String>,
 
     /// The identifier of the master core. This is the core the main thread
-    /// will run on. The default value is `0`.
+    /// will run on.
     pub master_core: CoreId,
```
Comment (zeeshanlakhani) [inline core/src/config.rs]: so, we want to force users to have to put `master_core` in their config (and not rely on defaults)?

Comment (drunkirishcoder) [reply core/src/config.rs]: yeah not clear what's better, so err on the side of being more explicit.

**Pull request merged**